Bomb weight

TC-DarkSlayer opened the pull request:
- Makes most bombs as heavy as a keg, giving them "medium weight" tag
- Frag grenades now properly shoot out shrapnel
- Fragmines, if too many within a certain radius, will now decay 
( prevents some intentional crashes with them, as you don't need 100 fragmines on one block )
( Decays if there are more than 5 within a radius of 4 blocks.  (32/8 = 4) This is to not actually nerf fragmine users

DARK-DarkSlayer committed:
Bomb Weight

- Added medium weight to most of the bombs as their purpose is for bombers/artillery
- Guided rockets damage buffed to take around 3-4 hits to destroy an armored bomber

DARK-DarkSlayer committed:
Frag Grenade

- Frag Grenades now shoot out shrapnel

DARK-DarkSlayer committed:
AntiCrash Fragmine Spam

- Prevents too many fragmines from all being at the same spot for too long which would result in a massive lag spike or possibly a crash
- Since it's TC, radius is 4 blocks (32/8), so that the actual fragmine players won't get nerfed as hard

Gingerbeard5773 commented on TFlippy_TerritoryControl_Core_Dev/Entities/DecayIfSpammed.as:
```diff
@@ -0,0 +1,36 @@
+#include "DecayCommon.as";
```
Instead of adding a whole new base file to TC it would be wiser to just copy the code and place it in the fragmine file, since that gives more room for editing and we wont ever have to worry about file bloat & long term support. Not to mention that by editing this file you change other blobs and could potentially (probably wont) have unintended effects.

Gingerbeard5773 replied:
Btw just thought of a perfect solution that wouldn't require decay at all. Simply add in collision for frag mines against other frag mines, so they cannot be stacked like before.

jammer312 replied:
Imagine reducing the lag by introducing extra collision checks

DARK-DarkSlayer committed:
Fragmine Anti-crash

- As per ginger's suggestion, fragmines now collide with each other, so they can't be stacked, so all of them can't explode at once

Gingerbeard5773 commented on TFlippy_TerritoryControl_Weapons_Dev/Entities/Weapons/Fragmine/FragMine.as:
```diff
@@ -144,7 +144,7 @@ bool explodeOnCollideWithBlob(CBlob@ this, CBlob@ blob)
 bool doesCollideWithBlob(CBlob@ this, CBlob@ blob)
 {
-	return blob.getShape().isStatic() && blob.isCollidable();
+	return blob.getShape().isStatic() && blob.isCollidable() blob.getName() == "fragmine";
```
This is missing an operator and will error! I assume you can see what is missing.

DARK-DarkSlayer committed:
Missing Operator

- Forgot to use an operator since I was manually copying it down

Vam-Jam commented on TFlippy_TerritoryControl_Weapons_Dev/Entities/Weapons/FragGrenade/FragGrenade.as:
```diff
@@ -63,6 +63,10 @@ void DoExplosion(CBlob@ this)
 		
 		LinearExplosion(this, dir, 1.0f + XORRandom(16) + (modifier * 4), 16 + XORRandom(24), 2, 0.10f, Hitters::explosion);
+
+		CBlob @blob = server_CreateBlob("shrapnel", this.getTeamNum(), this.getPosition()); // + Vec2f(16 - XORRandom(32), -10));
```
Needs to be wrapped in isServer

DARK-DarkSlayer committed:
isServer() Shrapnel

- Made it so if (isServer()) for shrapnel creation
